### PR TITLE
infra(release): publish release

### PR DESCRIPTION
**Pull request by ST-DDT**
This implements the forth/last step of our automated release pipeline.

Tasks:

- Checkout faker
- Install pnpm
- Setup node version
- Install Deps
- Gather release info
- Prepare Readme
- Publish to npm registry
- Set dist-tag
- Push to release branch (e.g. v9)

---

Demo-Workflow run:

- https://github.com/ST-DDT/faker/actions/runs/9768753513/job/26966889916

Demo Readme:

> ## 📙 API Documentation
>
> - [Getting Started Guide](https://v9.fakerjs.dev/guide/)
> - [API Reference](https://v9.fakerjs.dev/api/)
> 
> ## 🚀 Features

---

### Automated Release Pipeline PRs:

1. https://github.com/faker-js/faker/pull/2981
2. https://github.com/faker-js/faker/pull/2988
3. https://github.com/faker-js/faker/pull/2990
4. https://github.com/faker-js/faker/pull/2991

Supersedes (as requested):

- #2591

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    types: [published]
+
+permissions:
+  contents: write # to push the release branch
+
+jobs:
+  publish:
+    name: Publish Release
+    runs-on: ubuntu-latest
+    steps:
```
Comment (ST-DDT) [inline .github/workflows/publish-release.yml:15]:
```suggestion

    steps:
```

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        env:
+          CYPRESS_INSTALL_BINARY: 0
+
+      - name: Set latest/next dist-tag
+        run: |
+          pnpm dist-tag add @faker-js/faker@$RELEASE_VERSION $DIST_TAG
+
+      - name: Push to Release Branch
+        run: |
+          git push origin $GITHUB_REF_NAME:v$RELEASE_MAJOR
```
Comment (Shinigami92) [inline .github/workflows/publish-release.yml:66]: **question (if-minor):** why is this needed?

Comment (ST-DDT) [reply .github/workflows/publish-release.yml:66]: This pushes the `next` content to the `v9` branch.

Comment (Shinigami92) [reply .github/workflows/publish-release.yml:66]: 1. what happens when we switch to v10 but there is not v10 branch yet? will it create automatically?
2. will this be done also for non-stables? e.g. v9.1.0-alpha.0

Comment (ST-DDT) [reply .github/workflows/publish-release.yml:66]: > 1. what happens when we switch to v10 but there is not v10 branch yet? will it create automatically?

Yes

> 2. will this be done also for non-stables? e.g. v9.1.0-alpha.0

Yes, we didn't have that case before so I haven't thought about it.

**Review by Shinigami92 — APPROVED** [review]

File: .github/workflows/publish-release.yml (hunk 2 above)
Comment (ST-DDT) [reply .github/workflows/publish-release.yml:66]: Clarification: This would only be relevant for consideration if we do minor pre-releases.
(major pre-releases aren't affected, because there is no stable release that we could show instead.
We didn't have minor pre-releases before so I haven't implemented any checks for that.
If we ever do unstable minor releases, we can then decide whether the related branch should point to the stable release or the latest release (and either add the checks or do manual reverse pushes to that branch).

**Review by xDivisionByZerox — APPROVED** [review]
